src/rgw/rgw_sal_motr.cc

```diff
@@ -1328,13 +1328,24 @@ int MotrBucket::list(const DoutPrefixProvider *dpp, ListParams& params, int max,
         ent.decode(iter);
         rgw_obj_key key(ent.key);
         if (params.list_versions || ent.is_visible()) {
-          if (key.name == params.marker.name &&
+          if (key.name == params.marker.name) {
               // filter out versions which go before marker.instance
-              ((!null_ent.key.empty() && params.marker.instance == "null" &&
+              // null_ent is not yet populated
+              if (null_ent.key.name == "") {
+                 if ((params.marker.instance == "" && 
+                      key.instance == "") || 
+                      (key.instance != "" && 
+                      key.instance < params.marker.instance))
+                    continue;
+              } else {
+              // null_ent is populated
+              if ((params.marker.instance == "" &&
```

Comment:
@vastradparayya could you please explain why this check is required ?

Reply:
@priyanka25081999 we separated null_ent before populating and after. First if is for first time request where null_ent is not populated.